pkg/devspace/dependency/util/util.go

```diff
@@ -99,7 +99,7 @@ func DownloadDependency(ctx context.Context, workingDirectory string, source *la
 		_, statErr := os.Stat(localPath)
 
 		// Update dependency
-		if !source.DisablePull || statErr != nil {
```

Comment:
`!source.DisablePull` is almost always true, and the second part of the boolean check ignored, meaning the repo is always pulled